dotnet/samples/GettingStartedWithProcesses/Step04/Steps/ManagerAgentStep.cs

```diff
@@ -0,0 +1,146 @@
+        }
+
+        // Commit any changes to the chat history
+        await historyProvider.CommitAsync();
+
+        // Evaluate current intent
+        IntentResult intent = await IsRequestingUserInputAsync(kernel, history, logger);
+
+        string intentEventId =
+            intent.IsRequestingUserInput ?
+                AgentOrchestrationEvents.AgentResponded :
+                intent.IsWorking ?
+                    AgentOrchestrationEvents.AgentWorking :
+                    CommonEvents.UserInputComplete;
+
+        await context.EmitEventAsync(new() { Id = intentEventId });
+    }
+
+    [KernelFunction(Functions.InvokeGroup)]
+    public async Task InvokeGroupAsync(KernelProcessStepContext context, Kernel kernel)
+    {
+        // Get the chat history
+        IChatHistoryProvider historyProvider = kernel.GetHistory();
+        ChatHistory history = await historyProvider.GetHistoryAsync();
+
+        // Summarize the conversation with the user to use as input to the agent group
+        string summary = await kernel.SummarizeHistoryAsync(ReducerServiceKey, history);
+
+        await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.GroupInput, Data = summary });
+    }
+
+    [KernelFunction(Functions.ReceiveResponse)]
+    public async Task ReceiveResponseAsync(KernelProcessStepContext context, Kernel kernel, string response)
+    {
+        // Get the chat history
+        IChatHistoryProvider historyProvider = kernel.GetHistory();
+        ChatHistory history = await historyProvider.GetHistoryAsync();
+
+        // Proxy the inner response
+        ChatCompletionAgent agent = kernel.GetAgent<ChatCompletionAgent>(AgentServiceKey);
+        ChatMessageContent message = new(AuthorRole.Assistant, response) { AuthorName = agent.Name };
+        history.Add(message);
+
+        await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.AgentResponse, Data = message });
+
+        await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.AgentResponded });
+    }
+
+    private static async Task<IntentResult> IsRequestingUserInputAsync(Kernel kernel, ChatHistory history, ILogger logger)
+    {
+        ChatHistory localHistory =
+        [
+            new ChatMessageContent(AuthorRole.System, "Analyze the conversation and determine if user input is being solicited."),
+            .. history.TakeLast(1)
+        ];
+
+        IChatCompletionService service = kernel.GetRequiredService<IChatCompletionService>();
+
+        ChatMessageContent response = await service.GetChatMessageContentAsync(localHistory, new OpenAIPromptExecutionSettings { ResponseFormat = s_intentResponseFormat });
+        IntentResult intent = JsonSerializer.Deserialize<IntentResult>(response.ToString())!;
+
+        logger.LogTrace("{StepName} Response Intent - {IsRequestingUserInput}: {Rationale}", nameof(ManagerAgentStep), intent.IsRequestingUserInput, intent.Rationale);
```

Comment:
I'd add more logging so the user realizes clearly what is going on when looking at the logs/traces.

Reply:
Actually, no logs aren't being emitted for the sample.  There's a bunch of logging in both processes and agent frameworks for each invocation site.  I left some notes in the sample on how to enable logging, but the core output becomes engulfed in logging messages from the framework, which (imho) degrades the ability to decipher the sample output.  I included this as a reminder that any `KernelFunction` is able to take an `ILogger` or `ILoggerFactor` instance as a parameter and because this type of internal model output can be problematic if hidden from audit.